### ISSUE TITLE
CasSaml11Response code lines unused

### ISSUE THREAD

**Issue by Jetski5822** [apereo/dotnet-cas-client#9]
I was looking at CasSaml1Response.cs and noticed that these lines are not actually used...

var xmlReaderSettings = new XmlReaderSettings();
xmlReaderSettings.ConformanceLevel = ConformanceLevel.Auto;
xmlReaderSettings.IgnoreWhitespace = true;
xmlReaderSettings.IgnoreComments = true;
xmlReaderSettings.CloseInput = true;

....

string authMethod = SamlUtils.GetAttributeValue(authenticationStmtNode.Attributes, "AuthenticationMethod");

...

IList<string> authValues = new List<string>();
IDictionary<string, IList<string>> authenticationAttributes = new Dictionary<string, IList<string>>();
authValues.Add(authMethod);
authenticationAttributes.Add("samlAuthenticationStatement::authMethod", authValues);

Is this intentional? Or have I miss understood something?

**Comment by hokiecoder** [comment]
I have commented out [lines 115-119](https://github.com/apereo/dotnet-cas-client/blob/a100ed07bcf901a8c351abadb287c28b9f344f18/DotNetCasClient/Validation/CasSaml11Response.cs#L115-L119), [line 152](https://github.com/apereo/dotnet-cas-client/blob/a100ed07bcf901a8c351abadb287c28b9f344f18/DotNetCasClient/Validation/CasSaml11Response.cs#L152), and [lines 163-166](https://github.com/apereo/dotnet-cas-client/blob/a100ed07bcf901a8c351abadb287c28b9f344f18/DotNetCasClient/Validation/CasSaml11Response.cs#L163-L166) of [CasSaml1Response.cs](https://github.com/apereo/dotnet-cas-client/blob/a100ed07bcf901a8c351abadb287c28b9f344f18/DotNetCasClient/Validation/CasSaml11Response.cs). I then compiled the assembly and used it in a test project, using the Saml11 ticket validator, with success. I also tested with the role provider enabled without issue.

As **Jetski5822** pointed out, `xmlReaderSettings` and `authenticationAttributes` are never consumed anywhere in the `ProcessValidAssertion()` method, as far as we can tell. Perhaps these are artifacts left over from initial development? If a project maintainer could confirm these findings, I will create a pull request to remove these unused lines and we can close out this old issue?

**Comment by phantomtypist** [comment]
@hokiecoder @Jetski5822 

I think I'm going to take a glance over at the Java and PHP client repos to see what they are doing to get some insight as to what this code might have been used for at one point or intended to be used for.

**Comment by phantomtypist** [comment]
This is orphaned code that was in the repo, since it was in Git, all the way back to this files first commit by @serac in https://github.com/apereo/dotnet-cas-client/commit/50046c7a62e40fca3da415952701ea6dc90af494

I didn't do any further research, but it may have been for debugging XML purposes or something carried over from the Java CAS Client project.

I'm going to remove the orphaned code in `develop` since this doesn't warrant a hotfix release.  I'll be available in the next minor/major release.

**Comment by TheHokieCoder** [comment]
@phantomtypist I looked through the Java client code and came to my understanding that the handling of XML documents in .NET is quite different than in Java, so the code doesn't really have much comparison. However, I did look through the docs on the properties that are, in a futile effort, being set in this .NET client's code as mentioned by *Jetski5822*. I agree with your assessment that the code is orphaned as `XmlReaderSettings` is not even used by the `XmlDocument` class directly. It seems, to me, that it is useful for when XML is being read from a source with the intention of being (possibly) modified and stored again. But in the case of this project, the XML is just being parsed to create a SAML response object.

Just an FYI to concur with your choice!
